goreq.go

```diff
@@ -309,7 +315,7 @@ func (r Request) Do() (*Response, error) {
 
 	timeout := false
 	var timer *time.Timer
-	if r.Timeout > 0 {
+	if transport, ok := transport.(transportRequestCanceler); ok && r.Timeout > 0 {
```

Comment:
Is the `transportRequestCanceler` interface necessary?. Why can't we type assert against *http.Transport to call the `CancelRequest` function?